src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -245,6 +249,32 @@ protected string GetTimestampedLogFile(string logFile)
                     Path.GetExtension(logFile))).AddDoubleQuote();
         }
 
+        /// <summary>
+        /// This function will remove the unknown runsettings node from runsettings for old testhost who throws exception for unknown node.
+        /// </summary>
+        /// <param name="runsettingsXml">runsettings string</param>
+        /// <returns>runsetting after removing unrequired nodes</returns>
+        protected string RemoveNodesFromRunsettingsIfRequired(string runsettingsXml, ITestRunEventsHandler eventHandler)
```

Comment:
`RemoveNodesFromRunsettingsIfRequired(string runsettingsXml, Action<TestMessageLevel, string> logMessage)`

Refactor in ProxyExecutionManager to provide this function. Exception handling in PEM can use the same function.